### ISSUE TITLE
Support to create/add an image from raw `[]byte`

### ISSUE THREAD

**Issue by kravemir** [johnfercher/maroto#117]
**Is your feature request related to a problem? Please describe.**

Image is already (generated and) stored in memory as `[]byte`. Do I need to convert it to base64, or save to file, before add the image to pdf document?

**Describe the solution you'd like**

There should be support for creation of image from `[]byte` or, (maybe?) more generic way:

```golang
type ByteProvider interface {
    Get() []byte
}
```

**Describe alternatives you've considered**
Converting to base64, or saving to file. However,.. image is binary data, and `[]byte` seems like the most relevant data type to represent image raw data (as base64 conversion takes some time).

**Comment by johnfercher** [comment]
I think that we can have another method as ```BytesImage()```, but could you explain better your case? Are you reading an image from the disk and than converting to base64 to use the ```Base64Image()```? If yes, why don't use the ```FileImage()```?

**Comment by kravemir** [comment]
@johnfercher thank you for prompt response!

> ...  but could you explain better your case? Are you reading an image from the disk and than ...

I'm generating PDF from existing data structure, which provides `ImageContent() []byte` function, where source of `[]byte` is irrelevant (can be file, cached file reading, temporary/in-memory byte,.. or whatever).

However, I have already experimented with this library a bit, and I've hit some more limitations. So, I've decided to go with https://github.com/jung-kurt/gofpdf. And, I'm already building a wrapper around it to generate some layouts more easily. So, feature request is no longer relevant.
